### PR TITLE
Add a SearchAnchor example to experimental demo

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to add an example for `SearchAnchor` widget.

Screenshot 2023-04-03 at 4 33 53 PMScreenshot 2023-04-03 at 4 34 23 PM

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

### CODE REVIEW

File: experimental/material_3_demo/lib/component_screen.dart [hunk 1]
```diff
@@ -2182,6 +2182,101 @@ class _SlidersState extends State<Sliders> {
...
+                        style: TextStyle(color: Colors.grey)),
+                  )
+                ];
+              }
+              return getSuggestions(controller);
+            },
+          ),
+          const SizedBox(height: 20),
+          if (selectedColor == null)
+            const Text('Search a color')
```
Comment (HansMuller) [inline experimental/material_3_demo/lib/component_screen.dart:2271]:
```suggestion
            const Text('Select a color')
```

File: experimental/material_3_demo/lib/component_screen.dart [hunk 2]
```diff
@@ -2182,6 +2182,101 @@ class _SlidersState extends State<Sliders> {
...
+  @override
+  State<SearchAnchors> createState() => _SearchAnchorsState();
+}
+
+class _SearchAnchorsState extends State<SearchAnchors> {
+  String? selectedColor;
+  List<ColorItem> searchHistory = <ColorItem>[];
+
+  Iterable<Widget> getHistoryList(SearchController controller) {
+    return searchHistory.map((color) => ListTile(
```
Comment (HansMuller) [inline experimental/material_3_demo/lib/component_screen.dart:2197]: Two space indent though out this method and the next one

File: experimental/material_3_demo/lib/component_screen.dart [hunk 3]
```diff
@@ -2291,3 +2386,26 @@ class ComponentGroupDecoration extends StatelessWidget {
     );
   }
 }
+
+enum ColorItem {
```
Comment (HansMuller) [inline experimental/material_3_demo/lib/component_screen.dart:2390]: Presumable everything still works if there are more colors than can fit?

File: experimental/material_3_demo/lib/component_screen.dart [hunk 4]
```diff
@@ -2182,6 +2182,101 @@ class _SlidersState extends State<Sliders> {
...
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ComponentDecoration(
+      label: 'Search',
+      tooltipMessage: 'Use SearchAnchor or SearchAnchor.bar',
+      child: Column(
+        children: [
```
Comment (HansMuller) [inline experimental/material_3_demo/lib/component_screen.dart:2251]:
```suggestion
        children: <Widget>[
```

File: experimental/material_3_demo/lib/component_screen.dart [hunk 5]
```diff
@@ -2182,6 +2182,101 @@ class _SlidersState extends State<Sliders> {
...
+          SearchAnchor.bar(
+            barHintText: 'Search colors',
+            suggestionsBuilder: (context, controller) {
```
Comment (HansMuller) [inline experimental/material_3_demo/lib/component_screen.dart:2254]: The builder parameters need types

Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart:2254]: Seems there's a lint rule to ask us to remove the type for the arguments.
Screenshot 2023-04-03 at 4 59 39 PM

File: experimental/material_3_demo/lib/component_screen.dart (hunk 3 above)
Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart:2390]: Yes, then the result should be able to scroll.

File: experimental/material_3_demo/lib/component_screen.dart (hunk 2 above)
Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart:2197]: Yeah, the format is odd here but it is caused by the `dart format` command:(

File: experimental/material_3_demo/lib/component_screen.dart (hunk 4 above)
Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart]: Fixed!

File: experimental/material_3_demo/lib/component_screen.dart (hunk 1 above)
Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart]: Changed:)

File: experimental/material_3_demo/lib/component_screen.dart (hunk 5 above)
Comment (HansMuller) [reply experimental/material_3_demo/lib/component_screen.dart:2254]: I see; the samples repo plays by different rules. OK

File: experimental/material_3_demo/lib/component_screen.dart (hunk 2 above)
Comment (HansMuller) [reply experimental/material_3_demo/lib/component_screen.dart:2197]: OK, I see.

**Review by HansMuller — APPROVED** [review]
LGTM

File: experimental/material_3_demo/lib/component_screen.dart (hunk 5 above)
Comment (QuncCccccc) [reply experimental/material_3_demo/lib/component_screen.dart:2254]: Yes, and the interesting thing is in G3, I saw a lot of code which just skip type. So I guess our users also have different rules:)

**Pull request merged**